sdk/samples/iot/readme.md

```diff
@@ -1,29 +1,234 @@
+
+## Sample Descriptions
+This section provides an overview of the different samples available to run and what to expect from each.
+
+**IoT Hub C2D Sample**
+* *Executable:* `paho_iot_hub_c2d_sample`
+
+  This [sample](https://github.com/Azure/azure-sdk-for-c/blob/master/sdk/samples/iot/paho_iot_hub_c2d_sample.c) receives incoming cloud-to-device (C2D) messages invoked from the Azure IoT Hub. X509 self-certification is used.
+
+**IoT Hub Methods Sample**
+* *Executable:* `paho_iot_hub_methods_sample`
+
+  This [sample](https://github.com/Azure/azure-sdk-for-c/blob/master/sdk/samples/iot/paho_iot_hub_methods_sample.c) receives incoming method commands invoked from the the Azure IoT Hub.  X509 self-certification is used. A method named `ping` is supported, which if successful will return a json payload of the following:
+
+  ```json
+  {"response": "pong"}
+  ```
+
+**IoT Hub Telemetry Sample**
+* *Executable:* `paho_iot_hub_telemetry_sample`
+
+  This [sample](https://github.com/Azure/azure-sdk-for-c/blob/master/sdk/samples/iot/paho_iot_hub_telemetry_sample.c) sends five telemetry messages to the Azure IoT Hub. X509 self-certification is used.
+
+**IoT Hub SAS Telemetry Sample**
+* *Executable:* `paho_iot_hub_sas_telemetry_sample`
+
+  This [sample](https://github.com/Azure/azure-sdk-for-c/blob/master/sdk/samples/iot/paho_iot_hub_sas_telemetry_sample.c) sends five telemetry messages to the Azure IoT Hub. SAS certification is used.
+
+**IoT Hub Twin Sample**
+* *Executable:* `paho_iot_hub_twin_sample`
+
+  This [sample](https://github.com/Azure/azure-sdk-for-c/blob/master/sdk/samples/iot/paho_iot_hub_twin_sample.c) gets the twin document, reports its property, and receives desired property messages, all from or to the Azure IoT Hub. X509 self-certification is used.  A property named `device_count` is used for this purpose.
+
+* To send a device twin desired property message from the service to the device, open the device twin document in your Azure IoT Hub.  Add the property `device_count` along with a corresponding value to the `desired` section of the JSON.
+
+  ```json
+  {
+    "properties": {
+      "desired": {
+        "device_count": 42,
+      }
+    }
+  }
+  ```
+* Select Save to send the message. The device will store the value locally and report the updated property to the service.
+
+**IoT Hub PnP Sample**
+* *Executable:* `paho_iot_hub_pnp_sample`
+
+  Connect a PnP enabled device with the Digital Twin Model ID (DTMI) detailed [here](https://github.com/Azure/opendigitaltwins-dtdl/blob/master/DTDL/v2/samples/Thermostat.json). X509 self-certification is used.
+
+  In short, the capabilities are listed here:
+* **Methods**: Invoke a method called `getMaxMinReport` with JSON payload value `"since"` with an [ISO8601](https://en.wikipedia.org/wiki/ISO_8601) value for start time for the report. The method sends a response containing the following JSON payload:
```

Comment:
note there was a PR for a readme linter which means we are going to use `-` for unordered lists instead of `*`

Reply:
https://github.com/Azure/azure-sdk-for-c/pull/959

Reply:
Could you open the README in VSCode with [this](https://marketplace.visualstudio.com/items?itemName=DavidAnson.vscode-markdownlint) linter to match what Ahson merged?

Reply:
Another is `Lists should be surrounded by blank lines`

We may want to advocate use of such a linter team-wide. I was reluctant to add it to CI because of the false positives and exceptions, but maybe that's the way to go, to "stay green". This is part of this feature request: https://github.com/Azure/azure-sdk-for-c/issues/459

Reply:
Thanks all

Reply:
> note there was a PR for a readme linter which means we are going to use `-` for unordered lists instead of `*`

To clarify, I think the linter enforces local consistency. So, if the doc starts using `*` in lists, then it expects the whole doc to use it. If you start with `-`, it expects that going forward. Other than consistency, I don't think there is any intrinsic benefit of using one vs the other.